exporter/datadogexporter/config.go

```diff
@@ -412,6 +414,11 @@ type HostMetadataConfig struct {
 	// These tags will be attached to telemetry signals that have the host metadata hostname.
 	// To attach tags to telemetry signals regardless of the host, use a processor instead.
 	Tags []string `mapstructure:"tags"`
+
+	// sourceTimeout is the timeout to fetch from each provider - for example AWS IMDS.
+	// If unset, or set to zero duration, there will be no timeout applied.
+	// Default is 31 seconds.
+	sourceTimeout time.Duration `mapstructure:"source_timeout"`
```

Comment:
Struct tag is unnecessary and doesn't really work (Go reflection does not allow you to access unexported fields), so we can remove it
```suggestion
	sourceTimeout time.Duration
```